sdk/include/microvium/microvium_port.h

```diff
@@ -291,10 +294,70 @@ static uint16_t crc16(MVM_LONG_PTR_TYPE lp, uint16_t size) {
  * The `context` passed to these macros is whatever value that the host passes
  * to `mvm_restore`. It can be any value that fits in a pointer.
  */
-#define MVM_CONTEXTUAL_MALLOC(size, context) ({ Timeout t = {0, 0}; heap_allocate(&t, context, size); })
+#define MVM_CONTEXTUAL_MALLOC(size, context)                                   \
+	({                                                                         \
+		Timeout t = {0, 0};                                                    \
```

Comment:
I realize this hunk is just reformatting, but...

Should this tie into the "gas counter" below somehow?  I know that counts in units of VM instructions, and maybe we want to consider `malloc` to be O(1) within a VM instruction, so perhaps not, but OTOH maybe we want to tell `malloc` to give up after `gas_left * rough_time_per_vm_instr` time?

Reply:
I don't think there's a good way of doing that.  This is currently a non-blocking malloc and the VM will stop if it can't allocate.  We might want to provide a hook that allows users to specify a timeout to use (possibly make the context a pointer to a timeout and an allocation capability, so they can set a per-invocation cumulative timeout).